gensim/models/_fasttext_bin.py

```diff
@@ -225,19 +227,48 @@ def _load_matrix(fin, new_format=True):
 
     num_vectors, dim = _struct_unpack(fin, '@2q')
 
-    float_size = struct.calcsize('@f')
-    if float_size == 4:
-        dtype = np.dtype(np.float32)
-    elif float_size == 8:
-        dtype = np.dtype(np.float64)
-    else:
-        raise ValueError("Incompatible float size: %r" % float_size)
-
-    matrix = np.fromfile(fin, dtype=dtype, count=num_vectors * dim)
+    count = num_vectors * dim
+    matrix = _fromfile(fin, None, count)
```

Comment:
It's a bit better to have something like

```python
if fname.endswith("gz"):
    # this will be here until https://github.com/numpy/numpy/issues/13470 will be fixed
    <new_custom_code_same_as_in_pr>
else:
    <old_code>
```